components/suggestion/emoticon_provider.tsx

```diff
@@ -106,15 +115,16 @@ export default class EmoticonProvider extends Provider {
     //
     // For now, this behaviour and difference is by design.
     // See https://mattermost.atlassian.net/browse/MM-17320.
-    findAndSuggestEmojis(text, partialName, resultsCallback) {
-        const recentMatched = [];
-        const matched = [];
+    findAndSuggestEmojis(text: string, partialName: string, resultsCallback: (params: ResultCallbackParams) => void): void {
+        const recentMatched: SuggestEmoji[] = [];
+        const matched: SuggestEmoji[] = [];
         const state = store.getState();
-        const skintone = state.entities?.preferences?.myPreferences['emoji--emoji_skintone']?.value || 'default';
-        const emojiMap = getEmojiMap(state);
+        const skintone: string = state.entities?.preferences?.myPreferences['emoji--emoji_skintone']?.value || 'default';
+        const emojiMap: EmojiMap = getEmojiMap(state);
         const recentEmojis = getRecentEmojis(state);
 
         // Check for named emoji
+        // @ts-expect-error: Is this possible to fix without porting EmojiMap to Typescript?
```

Comment:
Didn't find a solution to this but maybe someone else can figure it out to get rid of @ts-expect-error

The error is
```
Type 'undefined' must have a '[Symbol.iterator]()' method that returns an iterator. (ts-error 2488)
```

Reply:
Added emoji_map.d.ts